src/vs/editor/common/model/bracketPairsTextModelPart/bracketPairsTree/bracketPairsTree.ts

```diff
@@ -105,11 +106,25 @@ export class BracketPairsTree extends Disposable {
 				lengthOfString(c.text)
 			);
 		}).reverse();
+		this.queueParsing(edits);
+	}
 
-		this.astWithTokens = this.parseDocumentFromTextBuffer(edits, this.astWithTokens, false);
-		if (this.initialAstWithoutTokens) {
-			this.initialAstWithoutTokens = this.parseDocumentFromTextBuffer(edits, this.initialAstWithoutTokens, false);
+	private queueParsing(edits: TextEditInfo[]): void {
+		// Queues parsing of the document so that it does not block editor input latency. The only
+		// downside to this is that changing a character adjacent to a bracket may appear white for
+		// a brief period.
+		this.parseQueue.push(edits);
+		this.parseTimer.cancelAndSet(() => this.doParse(), 0);
```

Comment:
We should remove this and call `doParse()` before all public methods of the class.